### DISCUSSION TITLE
Karavan Docker and Environment Variables

### DISCUSSION THREAD

**Discussion by Groumph** [apache/camel-karavan#1195] [Q&A] [answered]
I am attempting to use an environment variable in a route, {{env:MyEnvVar}}.  Currently just using the Web UI.

I have added that environment variable in my docker-compose and added a .env file.  

Where should I be setting the environment variable to work in devmode and builds?

The message I get is, "Property with key [env:MyEnvVar] using function [env] returned null value which is not allowed."

Any direction would be appreciated.

**Comment by Groumph** [comment]
Any suggestions from anyone?

I am trying to use {{env:MyEnvVar}} in Karavan for a connection string. 

Then outside Karavan I am running the built container with docker run -e MyEnvVar=blah builtImage. 

The environment variable is not getting used in the route. If I use a java process to get MyEnvVar with System.Getenv, it works. 

@mgubaidullin do you have any suggestions? Or anyone else?

**Comment by mgubaidullin** [comment, marked as answer]
Environment variable propagation is not yet implemented

**Reply by Groumph** [reply]
Thank you!
